jsonp-policy/src/main/java/io/apiman/plugins/jsonp_policy/http/ContentType.java

```diff
@@ -0,0 +1,89 @@
+package io.apiman.plugins.jsonp_policy.http;
+
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A value object that represents the Content-Type header.
+ * 
+ * @author Alexandre Kieling <alex.kieling@gmail.com>
+ */
+public class ContentType {
+
+	private static final Pattern DELIMITING_PATTERN = Pattern.compile(";");
+	private static final Pattern TYPE_SUBTYPE_PATTERN = Pattern
+			.compile("([^/]+/[^/]+)");
+	private static final Pattern PARAMETER_PATTERN = Pattern
+			.compile("([^=]+)=([^=]+)");
+	private String typeSubtype;
+	private SortedMap<String, String> parameters;
+
+	/**
+	 * Constructor that takes a Content-Type string.
+	 * @param str Content-Type string
+	 */
+	public ContentType(String str) {
+		String[] parts = DELIMITING_PATTERN.split(str);
+		Matcher matcher = TYPE_SUBTYPE_PATTERN.matcher(parts[0]);
+
+		if (matcher.find()) {
+			typeSubtype = matcher.group(1).trim();
```

Comment:
I'm on the fence about this.  Will give it some thought.  In the meantime I've been meaning to add an eclipse formatter file to the apiman repo.  Will try to remember to do that today.